test/e2e/stackdriver_monitoring.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	gcm "google.golang.org/api/monitoring/v3"
+)
+
+var _ = framework.KubeDescribe("Stackdriver metrics", func() {
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gke")
+	})
+
+	f := framework.NewDefaultFramework("stackdriver-monitoring")
+
+	It("should publish metrics to Stackdriver [Feature:StackdriverMonitoring][Flaky]", func() {
+		framework.Logf("Start creating Stackdriver monitoring test")
+		testStackdriverMonitoring(f)
+	})
+})
+
+var (
+	// Stackdriver container metrics, as descirbed here:
+	// https://cloud.google.com/monitoring/api/metrics#gcp-container
+	stackdriverMetrics = []string{
+		"uptime",
+		"cpu/reserved_cores",
+		"cpu/usage_time",
+		"cpu/utilization",
+		"memory/bytes_total",
+		"memory/bytes_used",
+		"memory/page_fault_count",
+		"disk/bytes_used",
+		"disk/bytes_total",
+	}
+
+	pollFrequency = time.Second * 5
+	pollTimeout   = time.Minute * 7
+
+	rcName            = "resource-consumer"
+	replicas          = 6
```

Comment:
Might be a good idea to set this number dynamically, depending on the cluster size